R/integrated_scores.R

```diff
@@ -26,20 +26,19 @@ weighted_survival_score = function(loss, truth, distribution, times, t_max, p_ma
     unique_times = .c_get_unique_times(truth[, "time"], times)
   }
 
-  # get the cdf
+  # get the cdf matrix (rows => times, cols => obs)
   if (inherits(distribution, "Distribution")) {
     cdf = as.matrix(distribution$cdf(unique_times))
   }
-  else if (inherits(distribution, "array") &
-           length(dim(distribution)) == 3) {
-    # 'distribution' is a survival 3d array so create an
-    # `Arrdist` using the 'median' curve
-    arrdistr = distr6::as.Distribution(1 - distribution, fun = "cdf",
-      decorators = c("CoreStatistics", "ExoticStatistics"))
-    cdf = as.matrix(arrdistr$cdf(unique_times))
-  } else { # 'distribution' is a survival 2d matrix
-    mtc = findInterval(unique_times, as.numeric(colnames(distribution)))
-    cdf = 1 - t(distribution[, mtc])
+  else if (inherits(distribution, "array")) {
+    if (length(dim(distribution)) == 3) {
+      # survival 3d array, extract median
+      surv_mat = .ext_surv_mat(arr = distribution, which.curve = 0.5)
```

Comment:
I guess I don't understand why you can't add a new parameter to this function and then add a new parameter to the measures? I agree median is a sensible default but there will be some cases (e.g., BART) where this may underestimate model performance if the last curve is the 'best'

Reply:
Oh sorry, I hadn't realized this, thanks for pointing out. Somehow I was thinking to pass it through the `score()` function. Let me try implement what you suggested.

Reply:
Implemented :)